lib/lookup.js

```diff
@@ -76,20 +76,25 @@ function resolve(context, next) {
       if (rep.stripAnsi) {
         context.module.stripAnsi = rep.stripAnsi;
       }
-      if (rep.replace) {
-        if (!rep.repo && !meta.repository) {
-          next(new Error('no-repository-field in package.json'));
-          return;
+      if (!rep.npm) {
+        if (typeof rep.replace === 'boolean' && rep.replace === false) {
```

Comment:
this should be hoisted outside of this if statement to avoid having to do the equality check twice in a row for `!rep.npm